packages/kotti-ui/source/kotti-button/KtButton.vue

```diff
@@ -105,119 +130,116 @@ export default {
 		padding-top: var(--unit-1);
 		padding-bottom: var(--unit-1);
 		line-height: var(--default-button-line-height);
-		&.icon {
+
+		&.kt-button--has-icon {
 			align-items: baseline;
 			text-align: left;
+
 			.yoco {
 				position: relative;
 				left: calc(var(--unit-1) * -1);
 			}
 		}
 	}
-}
-
-// Size modifiers
-.kt-button.sm {
-	height: var(--small-button-height);
-}
 
-.kt-button.lg {
-	height: var(--large-button-height);
-}
-
-// Color modifier
-/* stylelint-disable-next-line */
-.kt-button {
-	color: var(--button-main-color-dark);
-	background-color: var(--interactive-02);
-	border-color: var(--ui-02);
+	&--size {
+		&-small {
+			height: var(--small-button-height);
+		}
 
-	&:hover {
-		background-color: var(--button-main-color-light);
-		border-color: var(--button-main-color-light);
-	}
+		&-medium {
+			height: var(--default-button-height);
+		}
 
-	.kt-circle-loading {
-		border-bottom-color: var(--button-main-color-dark);
-		border-left-color: var(--button-main-color-dark);
+		&-large {
+			height: var(--large-button-height);
+		}
 	}
-}
 
-.kt-button.primary {
-	color: var(--text-04);
-	background-color: var(--button-main-color);
-	border-color: var(--button-main-color-dark);
+	&--type {
+		&-danger {
+			color: var(--danger);
 
-	&:hover {
-		background-color: var(--button-main-color-dark);
-	}
+			.kt-circle-loading {
```

Comment:
this dude seems to always inherit color - maybe instead of duplications, move it one level above and `color: inherit`

Reply:
It’s `border-color` though. I don’t think it’s easily possible to inherit this via CSS